python/lsst/daf/butler/core/formatter.py

```diff
@@ -42,21 +49,27 @@ class Formatter(metaclass=ABCMeta):
     are supported.
     """
 
+    def __init__(self, fileDescriptor):
+        if fileDescriptor is not None and not isinstance(fileDescriptor, FileDescriptor):
+            raise TypeError("File descriptor must be a FileDescriptor")
+        self._fileDescriptor = fileDescriptor
```

Comment:
Seems like this might as well be a public attribute.

Reply:
No. It's a read-only property. There's a getter further down.

Reply:
Meaning I want to guarantee that people can't change the file location once the object is constructed.